src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java

```diff
@@ -868,4 +871,37 @@ public void sendCommandTest(){
     assertEquals("PONG", SafeEncoder.encode((byte[]) jedis.sendCommand(PING)));
   }
 
+
+  @Test
+  public void encodeCompleteResponse(){
+    HashMap<String,String> entry = new HashMap<>();
+    entry.put("foo", "bar");
+    jedis.xadd( "mystream", StreamEntryID.NEW_ENTRY, entry );
+    String status = jedis.xgroupCreate("mystream", "mygroup", null, false);
+
+    Object obj = jedis.sendCommand(XINFO, "STREAM", "mystream");
+    List encodeObj =  (List)SafeEncoder.encodeObject(obj);
+
+    assertEquals( 14, encodeObj.size() );
+    assertEquals( "length", encodeObj.get(0) );
+    assertEquals( 1L, encodeObj.get(1) );
+
+    List entryAsList = new ArrayList(2);
+    entryAsList.add("foo");
+    entryAsList.add("bar");
+
+    assertEquals( entryAsList, ((List)encodeObj.get(11)).get(1) );
+
+    assertEquals("PONG", SafeEncoder.encodeObject(jedis.sendCommand(PING)));
+
+    entry.put("foo2", "bar2");
+    jedis.hset("hash:test:encode", entry);
+    encodeObj =   (List)SafeEncoder.encodeObject(jedis.sendCommand(HGETALL, "hash:test:encode"));
+
+    assertEquals( 4, encodeObj.size() );
+    assertEquals( "foo", encodeObj.get(0) );
+    assertEquals( "foo2", encodeObj.get(2) );
```

Comment:
The order of `foo` and `foo2` is not guaranteed. Use a different approach to test these.

Reply:
Oh yes, thanks